lib/internal/http2/util.js

```diff
@@ -349,10 +354,19 @@ function updateSettingsBuffer(settings) {
     settingsBuffer[IDX_SETTINGS_MAX_FRAME_SIZE] =
       settings.maxFrameSize;
   }
-  if (typeof settings.maxHeaderListSize === 'number') {
+  if (typeof settings.maxHeaderListSize === 'number' ||
+      typeof settings.maxHeaderSize === 'number') {
     flags |= (1 << IDX_SETTINGS_MAX_HEADER_LIST_SIZE);
+    if (settings.maxHeaderSize && settings.maxHeaderListSize) {
+      if (settings.maxHeaderSize !== settings.maxHeaderListSize) {
```

Comment:
```js
if (settings.maxHeaderListSize &&
 (settings.maxHeaderListSize ^ settings.maxHeaderSize))
```

Reply:
@rexagod could you review the changes again?